src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -771,9 +785,9 @@ static bool IsMatch(Document doc, string collection)
                 return true;
             }
         }
-
+        
         public IEnumerable<Document> GetDocumentsStartingWith(DocumentsOperationContext context, string idPrefix, string matches, string exclude, string startAfterId,
-            long start, long take, Reference<long> skip = null, DocumentFields fields = DocumentFields.All)
+            long start, long take, Reference<long> skip = null, DocumentFields fields = DocumentFields.All, Reference<long> unmatchedSkipped = null)
```

Comment:
Why adding additional parameter? We already have `Reference<long> skip` for that purpose, don't we?

Reply:
Sure, but it's used in a different case (with streams and replacing start parameter) and doesn't work in our case. We need only information when it doing skip for unmatched items so I introduced new parameter.